Source/ASButtonNode.mm

```diff
@@ -246,6 +274,7 @@ - (CGFloat)contentSpacing
 - (void)setContentSpacing:(CGFloat)contentSpacing
 {
   if (ASLockedSelfCompareAssign(_contentSpacing, contentSpacing)) {
+    [self updateYogaLayoutIfNeeded];
```

Comment:
This really reveals a difficulty with the yoga API today. I suppose the fix is to have a flag that represents whether your yoga tree is dirty, and go down the tree cleaning it before each yoga calculation.

This can stay for now though.

Reply:
Yep agree. I thought around how we can optimize it and if new infrastructure in place we can actually accomplish this.